src/App.js

```diff
@@ -8,46 +8,101 @@ import AddTodoForm from './addTodoForm';
 function App() {
 
   const [todoList, setTodoList] = useState([])
-  const initialList = JSON.parse(localStorage.getItem('savedTodoList')) || []
-  const [isloading,setIsLoading] = useState(true)
-
-  useEffect(() =>{
-     new Promise((resolve,reject)=>{
-      setTimeout(() => {
-        resolve({ data: { todoList: initialList } });
-      }, 2000);
-     })
-     .then(result =>{
-       setTodoList(result.data.todoList)
-       setIsLoading(false)
-     })
-  }, [])
+  // let initialList = JSON.parse(localStorage.getItem('savedTodoList')) ?? []
+  const [isloading, setIsLoading] = useState(true)
 
+  const fetchData = async () => {
+    const options = {
+      method: 'GET',
+      headers: { 'Authorization': `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}` }
+    }
+    const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`
+    try {
+      const response = await fetch(url, options)
 
-  useEffect(() => {
-    if (!isloading) {
-      localStorage.setItem('savedTodoList', JSON.stringify(todoList))
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
+      const data = await response.json();
+      const todos = data.records.map((todo) => {
+        return { title: todo.fields.Title, id: todo.id, createdAt: todo.createdTime }
+      })
+      const newTodos = todos.sort((a, b) => new Date(a.createdAt) - new Date(b.createdAt))
```

Comment:
Nitpick:
Array methods are chainable, therefore you can avoid creating another variable by chaining `.sort()` to `.map()`:
```js
const todos = data.records.map(...).sort(...)
```

Reply:
True that!